yt/geometry/particle_geometry_handler.py

```diff
@@ -46,6 +46,9 @@ def convert(self, unit):
         return self.dataset.conversion_factors[unit]
 
     def _setup_filenames(self):
+        if hasattr(self, "data_files"):
```

Comment:
I'm not sure why this is needed. Seems to me like the only way this attribute is already set is that the method was overriden in some subclass, in which case this is never called. Unless you except this method to be called multiple times downstream ? Since this is a "private" method, all calls should be within the codebase and not in user codes, so if there _are_ multiple calls indeed I think it could be considered a bug and fixed there instead of here.

Reply:
This is a good question. When developing this PR, I found a couple instances where this function does get called multiple times. Functionally, it's fine, but it can be a performance hit. This does not apply to this PR anymore (i.e., `_setup_filenames` only ever gets called once).

If I change this to raising an exception, there are ~3-4 tests in the test suite that will fail. So there is a bug that may require some refactoring. I can remove this and we can accept that small performance hit for now or I could file an issue for this to be revisited. What do you think?

Reply:
So if I'm reading this right, this line is actively silencing failing tests ?
Since the problem is older than this PR, keeping track of it as an issue would be a good first step, and I agree the problem doesn't _need_ to be addressed here.